apps/sms/js/startup.js

```diff
@@ -112,17 +112,19 @@ var Startup = {
       MessageManager.init();
       Navigation.init();
 
-      // If target panel is different from the default one, let's load remaining
-      // scripts as soon as possible, otherwise we can wait until first page of
-      // threads is loaded and rendered.
-      var panelName = Navigation.getPanelName();
-      if (panelName && panelName !== 'thread-list') {
-        // Initialize default panel only after target panel is ready.
+      var panelName = Navigation.getPanelName() || Navigation.defaultPanel;
+
+      // If initial panel is default one and app isn't run from notification,
+      // then just navigate to it, otherwise we can delay default panel
+      // initialization until we navigate to requested non-default panel.
+      if (panelName === Navigation.defaultPanel &&
+        !navigator.mozHasPendingMessage('notification')) {
+        Navigation.toPanel(panelName);
```

Comment:
In thins case I would prefer @azasypkin approach more because it's 2.2 blocker and Navigation might be replaced by NG, and I don't think we'll need toPanelFromHash since we revert the url for notification system message(and use of this function is also misleading because in the most of case we give the url with hash was not trying to navigate to the url, but just stop the navigation). 

I would also agree we should do some proper wrap up like isDefaultPanel for readability, and it might be a good chance to make `defaultPanel` into private once we expose these methods, but it's optional and up to you.